packages/vscode-extension/src/common/State.ts

```diff
@@ -12,15 +12,58 @@ export type WorkspaceConfiguration = {
   stopPreviousDevices: boolean;
 };
 
+export type EnvironmentDependency = "androidEmulator" | "xcode" | "nodejs";
+
+export type ApplicationDependency =
+  | "nodeVersion"
+  | "packageManager"
+  | "cocoaPods"
+  | "nodeModules"
+  | "ios"
+  | "android"
+  | "pods"
+  | "reactNative"
+  | "expo"
+  | "expoRouter"
+  | "storybook"
+  | "easCli";
+
+export type InstallationStatus = "installed" | "notInstalled" | "installing";
+
+export type DependencyStatus = {
+  status: InstallationStatus;
+  isOptional: boolean;
+  details?: string;
+};
+
+export type EnvironmentDependencyRecord = Partial<Record<EnvironmentDependency, DependencyStatus>>;
+export type ApplicationDependencyRecord = Partial<Record<ApplicationDependency, DependencyStatus>>;
+
+export type ProjectStore = {
+  applicationContext: ApplicationContextState;
+};
+
+export type ApplicationContextState = {
+  applicationDependencies: ApplicationDependencyRecord;
+};
```

Comment:
```suggestion
export type ProjectStore = {
  applicationDependencies: ApplicationDependencyRecord;
};
```
I don't think we need to mirror the way components compose in the extension code in the shape of the shared state.